content/en/docs/concepts/workloads/pods/ephemeral-containers.md

```diff
@@ -0,0 +1,219 @@
+
+## Uses for ephemeral containers
+
+Ephemeral containers are useful for interactive troubleshooting when `kubectl
+exec` is insufficient because a container has crashed or a container image
+doesn't include debugging utilities.
+
+In particular, [distroless images](https://github.com/GoogleContainerTools/distroless)
+enable you to deploy minimal container images that reduce attack surface
+and exposure to [CVEs](https://cve.mitre.org/). Since they do not include a
+shell or any debugging utilities, it's difficult to troubleshoot distroless
+images using `kubectl exec` alone.
+
+When using ephemeral containers, it's helpful to enable [process namespace
+sharing](/docs/tasks/configure-pod-container/share-process-namespace/) so
+you can view processes in other containers.
+
+### Examples
+
+{{< caution >}}
+The examples in this section require the `EphemeralContainers` [Feature
+Gate](/docs/reference/command-line-tools-reference/feature-gates.md) to be
+enabled and kubernetes client and server version of at least 1.16.
+{{< /caution >}}
+
+The examples in this section demonstrate how ephemeral containers appear in
+the API. For a more practical example of troubleshooting, see the [Debug a
+Running Pod](/docs/tasks/debug-application-cluster/debug-running-pod.md) task.
+
+Ephemeral containers are created using the `ephemeralcontainers` subresource
+of pod, which can be demonstrated using `kubectl --raw`. First describe
+the ephemeral container to add as an `EphemeralContainers` list:
+
+```json
+{
+    "apiVersion": "v1",
+    "kind": "EphemeralContainers",
+    "metadata": {
+            "name": "example-pod"
+    },
+    "ephemeralContainers": [{
+        "command": [
+            "sh"
+        ],
+        "image": "busybox",
+        "imagePullPolicy": "IfNotPresent",
+        "name": "debugger",
+        "stdin": true,
+        "tty": true,
+        "terminationMessagePolicy": "File"
+    }]
+}
+```
+
+To update the ephemeral containers of the already running `example-pod`:
+
+```shell
+kubectl replace --raw /api/v1/namespaces/default/pods/example-pod/ephemeralcontainers  -f ec.json
+```
+
+This will return the new list of ephemeral containers:
+
+```json
+{
+   "kind":"EphemeralContainers",
+   "apiVersion":"v1",
+   "metadata":{
+      "name":"example-pod",
+      "namespace":"default",
+      "selfLink":"/api/v1/namespaces/default/pods/example-pod/ephemeralcontainers",
+      "uid":"a14a6d9b-62f2-4119-9d8e-e2ed6bc3a47c",
+      "resourceVersion":"15886",
+      "creationTimestamp":"2019-08-29T06:41:42Z"
+   },
+   "ephemeralContainers":[
+      {
+         "name":"debugger",
+         "image":"busybox",
+         "command":[
+            "sh"
+         ],
+         "resources":{
+
+         },
+         "terminationMessagePolicy":"File",
+         "imagePullPolicy":"IfNotPresent",
+         "stdin":true,
+         "tty":true
+      }
+   ]
+}
+```
+
+The ephemeral container now appears in a `kubectl describe`:
+
```

Comment:
nit: reword line 157. The container appears in kubectl describe? Rather, the EC resource is created?

Reply:
Okay now. The phrase `now appears in a` seemed odd to me.